### PR TITLE
Adds status checks for integration test Redshift cluster

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR adds some extra checking to make sure the Redshift cluster is in the correct state when resuming and pausing it before/after the data integration tests.

See example github action here as a test: https://github.com/aqueducthq/aqueduct/actions/runs/4825449922/jobs/8596314618

## Related issue number (if any)
ENG 2776

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: .github/workflows/periodic-integration-tests.yml
Comment (saurav-c) [inline .github/workflows/periodic-integration-tests.yml:1]: Ignore these changes, I'll revert them once I run one more test

File: scripts/data/redshift_test.py [hunk 1]
```diff
@@ -1,36 +1,93 @@
...
 
-def resume_redshift(aws_access_key_id, aws_secret_access_key):
+def resume_redshift(aws_access_key_id, aws_secret_access_key, retry=0):
+    """
+    Resumes the test Redshift cluster.
+    """
     client = _create_client(aws_access_key_id, aws_secret_access_key)
-
     status = _get_cluster_status(client, CLUSTER_NAME)
+
```
Comment (kenxu95) [inline scripts/data/redshift_test.py:20]: Can we print the status here?

Comment (kenxu95) [reply scripts/data/redshift_test.py:20]: Better might be logging each case what we're going to try, whatever you prefer.

File: .github/workflows/periodic-integration-tests.yml [hunk 2]
```diff
@@ -214,42 +214,42 @@ jobs:
...
-        with:
-          status: ${{ job.status }}
-          notification_title: ""
-          message_format: "{emoji} *{workflow}* has {status_message}"
-          footer: "{run_url}"
-          notify_when: "failure,warnings"
-          mention_users: ${{ env.ONCALL_SLACK_MEMBER_ID }}
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.ACTION_MONITORING_SLACK }}
+      # - name: Get the Slack ID for the current oncall
```
Comment (kenxu95) [inline .github/workflows/periodic-integration-tests.yml:238]: It's probably good for us to set timeouts for both the setup and teardown steps here.

**Review by kenxu95 — APPROVED** [review]

**Pull request merged**